src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -1059,7 +1062,38 @@ export default abstract class ComparisonStore
             return Promise.resolve([]);
         },
     });
+    @computed get alterationEnrichmentRowData(): AlterationEnrichmentRow[] {
+        if (
+            this.alterationsEnrichmentData.isComplete &&
+            this.alterationsEnrichmentAnalysisGroups.isComplete
+        )
+            return getAlterationRowData(
+                this.alterationsEnrichmentData.result!,
+                this.resultsViewStore
+                    ? this.resultsViewStore.hugoGeneSymbols
+                    : [],
+                this.alterationsEnrichmentAnalysisGroups.result!
+            );
+        return [];
+    }
 
+    @computed get genesWithMaxFrequency(): AlterationEnrichmentRow[] {
+        if (
+            this.alterationsEnrichmentData.isComplete &&
+            this.alterationsEnrichmentAnalysisGroups.isComplete
+        ) {
+            const alterationRowData: AlterationEnrichmentRow[] = getAlterationRowData(
+                this.alterationsEnrichmentData.result!,
+                this.resultsViewStore
```

Comment:
it's strange that this store has a reference to resultsViewStore (sometimes).  We should try to fix that if we can.  It's odd that if there is not resultsviewstore we are passing it empty array.  Why doesn't it need genes?

Reply:
the function getAlterationRowData is also used in GenomicAlterations tab. We have passed genomic data to PathwayMapper whose format is AlterationEnrichmentRow. So, we have reused this function to obtain necessary data to be passed to PathwayMapper. So, I dont know why resultsViewStore is used.